### DISCUSSION TITLE
Merging linked PRs did not closes the associated issues

### DISCUSSION THREAD

**Discussion by Thomas-Boi** [community/community#27044] [Actions]
<p>Hello,</p>
<p>For some reason, the repo I’m maintaining will not automatically closes issues that were linked with a PR. I made sure that the PRs are properly merged into our default branch but it still doesn’t happen.</p>
<p>Here’s the <a href="https://github.com/devicons/devicon/issues/702" rel="noopener nofollow ugc">issue</a> in our repo. Here is its content:</p>
<hr>
<h3>
<a name="bug-1" class="anchor" href="#bug-1"></a>Bug</h3>
<p>Merging a linked PR into the <code>master</code> branch does not close its corresponding issue. See <a href="https://github.com/devicons/devicon/issues/592" rel="noopener nofollow ugc">#592</a> and confirm the PR is merged into <code>master</code> using this <a href="https://github.com/devicons/devicon/commits/master?after=9f4f5cdb393299a81125eb5127929ea7bfe42889+34&amp;branch=master" rel="noopener nofollow ugc">url</a> (just <code>Ctrl + F</code> and search for <code>raspberry</code> ). Also a few others like <a href="https://github.com/devicons/devicon/issues/582" rel="noopener nofollow ugc">#582</a> and <a href="https://github.com/devicons/devicon/issues/543" rel="noopener nofollow ugc">#543</a>.</p>
<h3>
<a name="how-to-replicate-the-bug-2" class="anchor" href="#how-to-replicate-the-bug-2"></a>How to replicate the bug</h3>
<p>Make an issue and make a PR. Link those together and merge the PR into <code>master</code> . The Issue will probably not close.</p>
<h3>
<a name="possible-fixessolutions-3" class="anchor" href="#possible-fixessolutions-3"></a>Possible Fixes/Solutions</h3>
<p>Not sure. Here’s what we’ve tried:</p>
<ul>
<li>Linking the issue using the GitHub website</li>
<li>Linking the issue by using keywords and mentioning the issue number in the comments (<a href="https://github.com/devicons/devicon/pull/691" rel="noopener nofollow ugc">new icon: Qt (original) #691</a>)</li>
<li>Searching on GitHub Community (found no issues similar to this)</li>
</ul>
<p>My next move will be making a thread about this on the Community website. Hopefully, we’ll get an answer there</p>
<hr>
<p>Note:</p>
<ul>
<li>
<p>For our workflow, we merge our PRs into a <code>develop</code> branch using squash merge. After a while, we then merge it into the <code>master</code> branch using a normal merge commit.</p>
</li>
<li>
<p>I’m not sure if it’s related. Some of the Issues (like <a href="https://github.com/devicons/devicon/issues/592" rel="noopener nofollow ugc">this</a>)will have icons that show the PR were merged:<br>
</p>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/d/6/d66e5a2ba15cf1aaf17d93f76e09c241bb4e480d.png" data-download-href="/uploads/short-url/uAWBEd2djZ0FxN2qQyoDfIq3Gxv.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/43018778/181101442-bd84ca2e-6ef9-4765-8baf-2c1afad2b3e8.png" alt="image" data-base62-sha1="uAWBEd2djZ0FxN2qQyoDfIq3Gxv" width="690" height="56" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/6/d66e5a2ba15cf1aaf17d93f76e09c241bb4e480d_2_690x56.png, https://aws1.discourse-cdn.com/github/original/3X/d/6/d66e5a2ba15cf1aaf17d93f76e09c241bb4e480d.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/6/d66e5a2ba15cf1aaf17d93f76e09c241bb4e480d.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/6/d66e5a2ba15cf1aaf17d93f76e09c241bb4e480d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1009×82 5.46 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
</li>
</ul>
<p>Others (like <a href="https://github.com/devicons/devicon/issues/582" rel="noopener nofollow ugc">this</a>) don’t:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/3/c36e57081b7f7bbe7e1904bbcda469be88240e4b.png" data-download-href="/uploads/short-url/rSRvyHwqpaBcSaHeSJ7FYW599fZ.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/43018778/181101443-eb97ba72-2c46-46cb-9762-d55f0e7c6096.png" alt="image" data-base62-sha1="rSRvyHwqpaBcSaHeSJ7FYW599fZ" width="690" height="39" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/3/c36e57081b7f7bbe7e1904bbcda469be88240e4b_2_690x39.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/3/c36e57081b7f7bbe7e1904bbcda469be88240e4b_2_1035x58.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/3/c36e57081b7f7bbe7e1904bbcda469be88240e4b.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/3/c36e57081b7f7bbe7e1904bbcda469be88240e4b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1202×68 4.31 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Both are issues that should have been closed automatically</p>

**Comment by airtower-luna** [comment]
<p>How were those PRs linked with the issue? I’m not seeing any “Closes: <span class="hashtag">#592</span>” or similar in the description, and at least for the one merged PR that’s listed as linked I also looked at the commits in it and didn’t see any such tag either.</p>

**Comment by Thomas-Boi** [comment]
<p>Hi,</p>
<p>From what I understand in this <a href="https://github.blog/2013-05-14-closing-issues-via-pull-requests/" rel="noopener nofollow ugc">link</a>, as long as “closes <span class="hashtag">#issue_number</span>” is in one of the PR comments, it should be linked. See also <a href="https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue" rel="noopener nofollow ugc">this</a>.</p>
<p>The contributor made sure to have the keyword within the description and as a commit message as wel</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I noticed the PRs for the issues you’ve linked were all merged into <code>develop</code>, which is not your default branch. That explains it. The PR must be merged into the default branch to close issues. :slightly_smiling_face:</p>

**Comment by Thomas-Boi** [comment]
<p>It’s a bit annoying that GitHub won’t close the issue later when the commits eventually get merged into <code>master</code>.</p>
<p>Thank you for your response though</p>

**Comment by airtower-luna** [comment]
<p>I assume that would work if you didn’t use squash merges. That way the commits with the “closes” tags would end up on <code>master</code>, which should close the issues, too.</p>
